sharing/filter_range_by_datetime_vars.py

```diff
@@ -0,0 +1,234 @@
+"""
+This module filters the data by a range between two datetime values.
+"""
+
+from typing import Any, Tuple, List, Union  # pylint: disable=import-error
+from datetime import datetime, timedelta  # pylint: disable=import-error
+import re
+import pandas as pd  # pylint: disable=import-error
+from pandas.core.frame import DataFrame  # pylint: disable=import-error
+from filter_single_value import filter_single_value  # pylint: disable=import-error
+
+
+def filter_range_by_datetime_vars(
```

Comment:
This function has too many unncessary parameters. It should take:
1. The rule value to check
2. The direction of the rule
3. The variable to check
4. The data

Same with the filter_range_by_numeric_vars